jelly/src/main/java/org/kohsuke/stapler/jelly/JellyClassLoaderTearOff.java

```diff
@@ -56,46 +55,43 @@ public JellyClassLoaderTearOff(MetaClassLoader owner) {
     }
 
     public TagLibrary getTagLibrary(String nsUri) {
-        LoadingCache<String,TagLibrary> m=null;
+        ConcurrentMap<String,TagLibrary> m=null;
         if(taglibs!=null)
             m = taglibs.get();
         if(m==null) {
-            m = CacheBuilder.newBuilder().build(new CacheLoader<String,TagLibrary>() {
-                public TagLibrary load(String nsUri) {
+            m = new ConcurrentHashMap<>();
+            taglibs = new WeakReference<>(m);
+        }
+        TagLibrary tl = m.computeIfAbsent(nsUri, key -> {
                     if(owner.parent!=null) {
                         // parent first
-                        TagLibrary tl = owner.parent.loadTearOff(JellyClassLoaderTearOff.class).getTagLibrary(nsUri);
-                        if(tl!=null)    return tl;
+                        TagLibrary taglib = owner.parent.loadTearOff(JellyClassLoaderTearOff.class).getTagLibrary(key);
+                        if(taglib!=null)    return taglib;
                     }
 
-                    String taglibBasePath = trimHeadSlash(nsUri);
+                    String taglibBasePath = trimHeadSlash(key);
```

Comment:
I think you can revert these lines, just use `nsUri`, and simply ignore the `key` arg. (When the second arg to `computeIfAbsent` is an inline lambda, the lambda arg is basically redundant. I wish there were an overload taking `Supplier` rather than `Function`—would be more convenient.)

Reply:
Done.